plumbing/transport/fetch_pack.go

```diff
@@ -128,6 +128,28 @@ func (r *UploadPackRequest) Have(h ...plumbing.Hash) {
 	r.Haves = append(r.Haves, h...)
 }
 
+func (r *UploadPackRequest) IsEmpty() bool {
+	if len(r.Haves) < len(r.Wants) {
+		return false
+	}
+
+	for _, h := range r.Wants {
+		found := false
+		for _, oh := range r.Haves {
+			if h == oh {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return false
+		}
+	}
+
+	return true
+}
+
```

Comment:
Probably unrelated with this PR, but something we must address at some point.

The `UploadPackRequest` should probably use a `ulreq.Ulreq`.